servicemanager/servicemanager.go

```diff
@@ -211,101 +182,46 @@ func (sm *ServiceManager) Close() {
 	close(sm.validateTTLStopChannel)
 }
 
-// InstallService install service to the system.
-func (sm *ServiceManager) InstallService(
-	newService ServiceInfo, imageURL string, fileInfo image.FileInfo,
-) (err error) {
-	return <-sm.actionHandler.Execute(newService.ServiceID,
-		func(id string) error {
-			return sm.doInstallService(newService, imageURL, fileInfo)
-		},
-	)
-}
-
-func (sm *ServiceManager) RestoreService(serviceID string) error {
-	return <-sm.actionHandler.Execute(serviceID,
-		func(id string) error {
-			return sm.doRestoreService(serviceID)
-		},
-	)
-}
-
-// RemoveService removes service from the system.
-func (sm *ServiceManager) RemoveService(serviceID string) (err error) {
-	return <-sm.actionHandler.Execute(serviceID,
-		func(id string) error {
-			return sm.doRemoveService(serviceID)
-		},
-	)
-}
-
-// GetAllServicesStatus gets all services status.
-func (sm *ServiceManager) GetAllServicesStatus() ([]ServiceInfo, error) {
-	services, err := sm.serviceInfoProvider.GetLatestVersionServices()
-	return services, aoserrors.Wrap(err)
-}
-
 // GetServiceInfo gets service information by id.
 func (sm *ServiceManager) GetServiceInfo(serviceID string) (serviceInfo ServiceInfo, err error) {
-	if serviceInfo, err = sm.serviceInfoProvider.GetService(serviceID); err != nil {
+	services, err := sm.serviceInfoProvider.GetService(serviceID)
+	if err != nil {
 		return serviceInfo, aoserrors.Wrap(err)
 	}
 
-	if serviceInfo.Cached {
-		return serviceInfo, ErrNotExist
+	for _, service := range services {
+		if service.ServiceID == serviceID && !service.Cached {
+			return service, nil
+		}
 	}
 
-	return serviceInfo, nil
+	return serviceInfo, ErrNotExist
 }
 
 // GetImageParts gets image parts for the service.
 func (sm *ServiceManager) GetImageParts(service ServiceInfo) (parts ImageParts, err error) {
 	return getImageParts(service.ImagePath)
 }
 
-// ApplyService applies already installed service.
-func (sm *ServiceManager) ApplyService(service ServiceInfo) (err error) {
-	return <-sm.actionHandler.Execute(service.ServiceID,
-		func(id string) error {
-			return sm.doApplyService(service)
-		},
-	)
-}
-
-// RevertService reverts already installed service.
-func (sm *ServiceManager) RevertService(service ServiceInfo) (retErr error) {
-	return <-sm.actionHandler.Execute(service.ServiceID,
-		func(id string) error {
-			return sm.doRevertService(service)
-		},
-	)
-}
-
-// UseService sets last use of service.
-func (sm *ServiceManager) UseService(serviceID string, aosVersion uint64) error {
-	if err := sm.serviceInfoProvider.SetServiceTimestamp(serviceID, aosVersion, time.Now().UTC()); err != nil {
-		return aoserrors.Wrap(err)
-	}
-
-	serviceInfo, err := sm.serviceInfoProvider.GetService(serviceID)
+// ProcessDesiredLayers installs, removes, restores desired services on the system.
```

Comment:
// incorrect comment should be ProcesdDesiredServices

Reply:
Done